internal/services/manager_test.go

```diff
@@ -36,14 +44,36 @@ func TestNewManager(t *testing.T) {
 				t.Setenv("DBUS_SYSTEM_BUS_ADDRESS", tc.systemBusSocket)
 			}
 
+			destCmdsFile := filepath.Join(t.TempDir(), "gpasswd.output")
+			if tc.cleanupInterval != 0 {
+				groupFilePath := filepath.Join(testutils.TestFamilyPath(t), tc.groupsFile)
+				cmdArgs := []string{"env", "GO_WANT_HELPER_PROCESS=1",
+					fmt.Sprintf("GO_WANT_HELPER_PROCESS_DEST=%s", destCmdsFile),
+					fmt.Sprintf("GO_WANT_HELPER_PROCESS_GROUPFILE=%s", groupFilePath),
+					os.Args[0], "-test.run=TestMockgpasswd", "--"}
+				usertests.OverrideDefaultOptions(t, groupFilePath, cmdArgs)
+				servicestests.OverrideCleanupInterval(t, tc.cleanupInterval)
+			}
+
 			m, err := services.NewManager(context.Background(), tc.cacheDir, t.TempDir(), nil)
 			if tc.wantErr {
 				require.Error(t, err, "NewManager should have returned an error, but did not")
 				return
 			}
-			defer require.NoError(t, m.Stop(), "Teardown: Stop should not have returned an error, but did")
-
 			require.NoError(t, err, "NewManager should not have returned an error, but did")
+
+			if tc.cleanupInterval > 0 {
+				// Sleep for a while to ensure the cleanup interval is triggered.
+				time.Sleep(tc.cleanupInterval + time.Second)
```

Comment:
Can't we be less time-depending here?

Reply:
In this case, I don't think so. The cleanup is triggered periodically, so it's important to depend on the time here.